### PR TITLE
DashboardAI: UX improvements

### LINKED ISSUE #84213: UX improvements on "improve" toggletip
**Content generated**
- Change the placement of the content generated to be at the top

**History arrows (counter)**
- Move the placement of **history arrows** to be below on the **bottom-right** to the content generated

**Actions**
- Move the placement for the **quick actions** _(Event shorter/More descriptive/Regenerate)_ to be next to the main CTA, below the content generated
- Add **leading icons** for the quick actions buttons
- Change the wording of the **"Apply**" button to **"Apply AI text"**. Change the size to be "**sm**" as the other quick actions buttons

**Input**
- Change the placement of the `input` to be at the bottom
- Size of the **height** of the `input` to be `32px` with `8px` of **padding**
- Change the icon to send the message input, to a "**ghost button**" with a leading icon and a label that says **"Send"**

**Information for AI documentation**
- Change the style of the **text-link** of the "Grafana LLM plugin" link to follow our **[Saga DS guidelines](https://grafana.com/developers/saga/components/text-link)**

[Figma link](https://www.figma.com/file/yBuMIJDWHJbsfujZK2raRo/DashGPT?type=design&node-id=986-8645&mode=design)

Toggletip_AI_text_generation

cc @ivanortegaalba @nmarrs @adela-almasan @SandersAaronD

_Originally posted by @natydej in https://github.com/grafana/grafana/issues/83925#issuecomment-1984250340_

### PR DESCRIPTION
**Pull request by adela-almasan**
dashgpt_ux

- [x] Move generated content at the top 
- [x] Move arrows below the generated content
- [x] Move Quick actions buttons next to main CTA button below generated content
- [x] Add icons to Quick actions buttons
- [x] Apply -> Apply AI text + update size
- [x] Input at the bottom
- [x] Update Icon to "enter" + Send
- [x] Size of the height of the input to be 32px with 8px of padding (already done?)
- [ ] Change the style of the text-link of the "Grafana LLM plugin" link to follow our [Saga DS guidelines](https://grafana.com/developers/saga/components/text-link)

Fixes #84213

**Special notes for your reviewer:**

- Also fixes bug where the messages were empty in history, leading to inaccurate re-generations.
- Looks like the toggletip text color is overriding the `TextLink` styling

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by natydej** [comment]
Looks great @adela-almasan 
Just one thing, for the quick actions buttons, not sure why the `gap` changed from **8px** to **4px**, I think we should keep it as **8px**.

**Comment by adela-almasan** [comment]
> Looks great @adela-almasan Just one thing, for the quick actions buttons, not sure why the `gap` changed from **8px** to **4px**, I think we should keep it as **8px**.

@natydej not sure either, but I changed it back.

**Comment by ivanortegaalba** [comment]
@Ijin08 @imatwawana I'd love to ask you about the copies. What do you think?

dashgpt_ux

**Comment by ivanortegaalba** [comment]
I'm not very convinced about the "Apply AI text" button copy. Maybe, `Apply` is enough.

**Review by ivanortegaalba — APPROVED** [review]
I asked for feedback to @Ijin08 and @imatwawana about the copies. But codewise looks great

File: go.work.sum
Comment (ivanortegaalba) [inline go.work.sum:1]: This happened to me today too, why do we get these changes? 🤔 Probably we can skip those from the PR

Comment (adela-almasan) [reply go.work.sum:1]: not sure, apparently today was not a great day for that file. yeah, I'll remove the changes

**Comment by nmarrs** [comment]
> I'm not very convinced about the "Apply AI text" button copy. Maybe, `Apply` is enough.

I agree with this feedback

**Review by nmarrs — APPROVED** [review]
Tested locally and this is looking great! just a few minor copy nits but IMO this is good to go 😄💯

**Comment by Ijin08** [comment]
Some general thoughts:
- Something with the structure is of and I think it is the placement of the apply button. Would expect it to be furthest down above the footer. Right now for me it seems the input is for a completely unrelated task.
- I would rename the submit button to "Apply" or "Apply title". The title is what the user wants. The footer is enough to indicate that AI is being used. The user doesn't want a AI generated title. They just want help with generating a title.
- The "Even shorter" etc buttons are quick actions for the input so should be more connected to it.

Here is a rough idea of what I mean:
Screenshot 2024-03-22 at 14 05 14

**Comment by imatwawana** [comment]
I'll wait to make any suggestions on copy until @Ijin08's comments have been addressed.

**Comment by adela-almasan** [comment]
take 2 - @ivanortegaalba, @Ijin08, @natydej, @nmarrs 
Screenshot 2024-03-22 at 12 12 12 PM

**Comment by natydej** [comment]
@adela-almasan Looks good to me! with the quick actions and the "Apply" button being separated as Patrick suggested.

**Review by natydej — APPROVED** [review]
LGMT!

**Review by nmarrs — APPROVED** [review]
LGTM (again) 😄

**Comment by ivanortegaalba** [comment]
This looks amazing @adela-almasan!!! Thanks everyone for the good feedback! 🚀

**Pull request merged**